### ISSUE TITLE
Trait by latitude plots

### ISSUE THREAD

**Issue by DeirdreLoughnan** [DeirdreLoughnan/Treetraits#10]
@lizzieinvancouver the more I think about how I made the plots depicting the change in traits with latitude the less I think they are right:

![Screenshot 2024-12-19 at 8 44 08 PM](https://github.com/user-attachments/assets/8ea408c0-7a92-4aac-ac8b-d1769d039773)

My approach was to simulate trait values based on the mean and 90% UI for the model parameters from the trait part of the model:

```
# extract parameter estimates
sumHt <- summary(mdlHt)$summary[c("mu_grand", "b_tranE", "b_tranlat"),]
intHtQuant <- quantile(postHt$mu_grand, c(0.05, 0.95))
b_tranHtQuant <- quantile(postHt$b_tranE, c(0.05, 0.95))
b_tranlatHtQuant <- quantile(postHt$b_tranlat, c(0.05, 0.95))

#simulate values
lati <- seq(40, 60, by = 0.5)
tranW <- 0
tranE <- 1

#simulate trait estimates
ht_w = sumHt["mu_grand",1] + sumHt["b_tranE",1] * tranW + sumHt["b_tranlat",1] * (tranW*lati)
ht_e = sumHt["mu_grand",1] + sumHt["b_tranE",1] * tranE + sumHt["b_tranlat",1] * (tranE*lati)

ht_w5 = intHtQuant[1] + b_tranHtQuant[1] * tranW + b_tranlatHtQuant[1] * (tranW*lati)
ht_e5 = intHtQuant[1] + b_tranHtQuant[1] * tranE + b_tranlatHtQuant[1] * (tranE*lati)

ht_w95 = intHtQuant[2] + b_tranHtQuant[2] * tranW + b_tranlatHtQuant[2] * (tranW*lati)
ht_e95 = intHtQuant[2] + b_tranHtQuant[2] * tranE + b_tranlatHtQuant[2] * (tranE*lati)

htEW <- data.frame(htw = ht_w, hte = ht_e,  ht_w5 =ht_w5, ht_w95 = ht_w95, ht_e5 = ht_e5, ht_e95 = ht_e95  )

# plotting code:
intHt <- ggplot(htEW) +
  geom_line(aes(y = htw, x = lati), col = "cyan4", lty = "dashed") +
  geom_ribbon(data = htEW, aes(ymin = ht_w5, ymax = ht_w95, x= lati), alpha = 0.2, fill = "cyan4") +
  geom_line(aes(y = hte, x = lati), col = "cyan3") + 
  geom_ribbon(data = htEW, aes(ymin = ht_e5, ymax = ht_e95, x= lati), alpha = 0.2, fill = "cyan3") + 
  xlab("Latitude") + ylab("Height (m)") 
  
```

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver I tweaked the code a bit to make it more readable. The new version is in the git issue.

I also recall you saying you would need to see the sigmas, so here is a subset of the summary statistics from the trait model:
```
                   mean       2.5%        25%        50%        75%       97.5%
mu_grand      5.3502067  4.0908419  4.9155266  5.3467523  5.7801286  6.65575193
b_tranE       4.9831156 -4.3849740  1.8073047  5.0637911  8.1558375 14.27413335
b_tranlat    -0.1496195 -0.3576397 -0.2212631 -0.1511528 -0.0782439  0.05984286
sigma_traity  2.8720683  2.7620464  2.8327207  2.8711489  2.9103022  2.98603524
sigma_sp      4.2905201  3.4527154  3.9408867  4.2581635  4.5879797  5.35610972
```

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Can you ... (no rush, I am pretty underwater):

1. Remind me of the full equation for the model you're fitting? I can guess from the issue but better if it is spelled out here. 
2. Remind me why you would use the mean and quantiles to simulate versus the full posterior of each parameter?

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver 

1. The model equations are:
```
\\traits 

 for(i in 1:n_spec){
    mu_grand_sp[i] = mu_grand + muSp[i];
  }
  
  for (i in 1:N){
    y_hat[i] =  mu_grand + muSp[trait_species[i]] + b_tranE * tranE[i]  + b_tranlat * interTranLat[i] 
    ;
  }
  
  // Phenology
  for (isp in 1:n_spec){
    betaForceSp[isp] = alphaForceSp[isp] + betaTraitxForce * (mu_grand_sp[isp]);
  }
  for (isp in 1:n_spec){
    betaPhotoSp[isp] = alphaPhotoSp[isp] + betaTraitxPhoto * (mu_grand_sp[isp]);
  }
  for (isp in 1:n_spec){
    betaChillSp[isp] = alphaChillSp[isp] + betaTraitxChill * (mu_grand_sp[isp]);
  }
  
}

//Phenology

for (i in 1:Nph){
    yPhenoi[i] ~ normal(alphaPhenoSp[phenology_species[i]] + betaForceSp[phenology_species[i]] * forcei[i] + betaPhotoSp[phenology_species[i]] * photoi[i] + betaChillSp[phenology_species[i]] * chilli[i], sigmapheno_y);
  }
```


2. I guess at the time I thought they would be functionally similar, but I can see how that is not true. By simulating across the full posterior, do you mean something like:

```
lati <- seq(40, 60, length.out = 25)

output <- data.frame(cbind(
  rep(1:(length(lati)*nrow(postHt))), 
  rep(postHt$mu_grand, times = length(lati)), 
  rep(postHt$b_tranE, times = length(lati)),
  rep(postHt$b_tranlat, times = length(lati)),
  rep(lati, each = 1000)))

names(output) <- c("iter", "mu_grand","b_tranE", "b_tranLat","lat")
output$latTrend <- NA  

for (i in 1:nrow(output)){
  temp <- output$mu_grand[i] + output$b_tranE[i]* tranE + output$b_tranLat[i]* (tranE*output$lat[i])
  output$latTrend[output$iter == i] <- temp
}

```
Which produced this figure for height (pink is the western transect and blue eastern)

![Screenshot 2025-01-03 at 10 41 54 PM](https://github.com/user-attachments/assets/838c001a-c10b-4628-9484-1d4c7f0a5569)

**Comment by lizzieinvancouver** [comment]
> By simulating across the full posterior, do you mean something like:

@DeirdreLoughnan Yes! Just take the full posteriors and run the equation from the model using the full posterior for each parameter, then vary your input to turn on or off each transect and look at the effect across latitudes .... which looks like what your code does to me (but double check me). 

That said, do you allow the effect of latitude on height to vary by transect? Your plot suggests you do, but I don't quite see it here:

`    y_hat[i] =  mu_grand + muSp[trait_species[i]] + b_tranE * tranE[i]  + b_tranlat * interTranLat[i] 
`

Or maybe I am missing something?

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver I have been working on updating this figure and here is the new version:

<img width="1303" alt="Image" src="https://github.com/user-attachments/assets/5362a4b2-2bd0-45ed-8fea-08f75a12e871" />

> That said, do you allow the effect of latitude on height to vary by transect? Your plot suggests you do, but I don't quite see it here:
> 
> ` y_hat[i] = mu_grand + muSp[trait_species[i]] + b_tranE * tranE[i] + b_tranlat * interTranLat[i] `
> 
> Or maybe I am missing something?

Transect was included as a dummy variable (0,1), with the western transect being the 0. But let me know if this is something we should discuss further.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan I like the look of the figure (though the y axes need to be scaled better to each trait) but I cannot interpret it so easily. Can you add a legend and also the effects from the model printed out (summary of the fit basically)?

I suspect your model is okay, but cannot read through the names of the objects (tranE and interTranLat). Can you define those here and also give their range of values? Thanks!

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver here is a quick reply, but perhaps we can discuss more Wednesday what you mean by "effects from the model" if necessary. 

Here is a more zoomed in version of the above figure, but rescaling LMA and WSD for which I had to multiply by 100 to get the model to work (as discussed in issue #6 and #19).

<img width="1434" alt="Image" src="https://github.com/user-attachments/assets/c7b296ad-6ee8-4ec4-819c-3ffb128b78be" />

```
bb = mu_grand + muSp[trait_species[i]] + b_transect * transect(0/1) + b_transect by latitude* (transect(0/1) * latitude)

```

Where the western transect is coded as "0" and the eastern transect as "1" and latitude is a continuous variable (ranging from 42.53-54.78 degree latitude). 

The mean model estimated values (summary values) are:
```
trait        mu.grand    b_tran b_tranxlat
Height    5.35            4.98     -0.15    
DBH       6.20          -32.00      0.67
WSD*      0.49            0.54     -0.01
LMA*      0.038           -0.14      0.003
LNC       3.72            2.31     -0.09
```
*Values are rescaled

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Happy to talk Wednesday! 

By 'effects from the model' I meant the summary output you added, though it would be helpful to have some uncertainty added. The means don't give much insight into whether the figures look right .... 

Can you confirm your model has no main effect for latitude? If it doesn't that's a little weird, if it does, I am not sure you're plotting it here (and you should).
